src/java.base/share/classes/com/sun/crypto/provider/SunJCE.java

```diff
@@ -819,4 +849,56 @@ static SunJCE getInstance() {
         }
         return instance;
     }
+
+
+    static {
+
+        String nativeCryptTrace = GetPropertyAction.privilegedGetProperty("jdk.nativeCryptoTrace");
+        String nativeCryptStr = GetPropertyAction.privilegedGetProperty("jdk.nativeCrypto");
+        String nativeChaCha20Str = GetPropertyAction.privilegedGetProperty("jdk.nativeChaCha20");
+
+        if (Boolean.parseBoolean(nativeCryptStr) || nativeCryptStr == null) {
```

Comment:
The null check should occur first: `(nativeCryptStr == null) || Boolean.parseBoolean(nativeCryptStr)`

Reply:
fixed (and fixed the other files that have this check)